### PR TITLE
ADR-994: Spirits journey- alcohol used screen changes

### PR DESCRIPTION
**Pull request by sini-george**

### CODE REVIEW

File: app/forms/spiritsQuestions/AlcoholUsedFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      "beer"         -> bigDecimal(
+        quantityMaxDecimalPlaces,
+        "alcoholUsed.error.beer.required",
+        "alcoholUsed.error.beer.nonNumeric",
+        "alcoholUsed.error.beer.twoDecimalPlaces"
+      )
+        .verifying(minimumValue(quantityMinValue, "alcoholUsed.error.beer.minimumRequired"))
+        .verifying(maximumValue(quantityMaxValue, "alcoholUsed.error.beer.maximumRequired")),
+      "wine"         -> bigDecimal(
+        2,
```
Comment (msatta) [inline app/forms/spiritsQuestions/AlcoholUsedFormProvider.scala:39]: Should we use quantityMaxDecimalPlaces here?

File: test/forms/spiritsQuestions/AlcoholUsedFormProviderSpec.scala [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import scala.collection.compat.immutable.ArraySeq
+
+class AlcoholUsedFormProviderSpec extends BigDecimalFieldBehaviours {
+
+  val form = new AlcoholUsedFormProvider()()
+
+  ".beer" - {
+
+    val fieldName   = "beer"
+    val requiredKey = "alcoholUsed.error.beer.required"
```
Comment (msatta) [inline test/forms/spiritsQuestions/AlcoholUsedFormProviderSpec.scala:31]: What requiredKey has done to you? Why the special treatment? 😄 Could we have the same implementation for nonNumeric and twoDecimalPlaces too?

File: app/forms/spiritsQuestions/AlcoholUsedFormProvider.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        "alcoholUsed.error.ciderOrPerry.required",
+        "alcoholUsed.error.ciderOrPerry.nonNumeric",
+        "alcoholUsed.error.ciderOrPerry.twoDecimalPlaces"
+      )
+        .verifying(minimumValue(quantityMinValue, "alcoholUsed.error.ciderOrPerry.minimumRequired"))
+        .verifying(maximumValue(quantityMaxValue, "alcoholUsed.error.ciderOrPerry.maximumRequired"))
+    )(AlcoholUsed.apply)(AlcoholUsed.unapply)
+  )
+}
+object AlcoholUsedFormProvider {
```
Comment (msatta) [inline app/forms/spiritsQuestions/AlcoholUsedFormProvider.scala:65]: Why use a companion object if the only place we use them is inside the class?

File: test/forms/spiritsQuestions/AlcoholUsedFormProviderSpec.scala (hunk 2 above)
Comment (sini-george) [reply test/forms/spiritsQuestions/AlcoholUsedFormProviderSpec.scala]: Fixed this one

**Review by msatta — APPROVED** [review]

**Pull request merged**